src/Utilities/PrettyType.hpp

```diff
@@ -705,4 +706,26 @@ template <typename T>
 std::string short_name() {
   return detail::extract_short_name(typeid(T).name());
 }
+
+namespace detail {
+template <typename T, typename = std::void_t<>>
+struct name_helper {
+  static std::string name() { return pretty_type::short_name<T>(); }
+};
+
+template <typename T>
+struct name_helper<T, std::void_t<decltype(T::name())>> {
+  static std::string name() { return T::name(); }
+};
+}  // namespace detail
+
+/*!
+ * \ingroup PrettyTypeGroup
+ * \brief Return the result of the `name()` member of a class. If a class
+ * doesn't have a `name()` member, call `pretty_type::short_name<T>()` instead.
+ */
+template <typename T>
+std::string name() {
```

Comment:
What do you think about moving this into a shorter namespace such as `tt` now that it is used so much throughout the code, and to avoid confusion with `short_name? Should be a fairly straightforward find-replace right? Get a second reviewer's opinion before doing it.

Reply:
I don't understand how this `name()` function is a type trait. Why would it make sense to move a non-type trait thing into the type trait namespace? I think `pretty_type` is the right place for this function

Reply:
I agree with @nilsdeppe.